snap/local/hooks/cmd/install/install.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package main
+
+import (
+	"os"
+
+	hooks "github.com/canonical/edgex-snap-hooks/v2"
+	"github.com/canonical/edgex-snap-hooks/v2/env"
+	"github.com/canonical/edgex-snap-hooks/v2/log"
+)
+
+// installProfiles copies the profile configuration.toml files from $SNAP to $SNAP_DATA.
+func installConfig() error {
+	resPath := "/config/device-usb-camera/res"
+	err := os.MkdirAll(env.SnapData+resPath, 0755)
+	if err != nil {
+		return err
+	}
+
+	path := resPath + "/configuration.toml"
+	err = hooks.CopyFile(
+		env.Snap+path,
+		env.SnapData+path)
+	if err != nil {
+		return err
+	}
+
+	path = "/config/rtsp-simple-server.yml"
+	err = hooks.CopyFile(
+		env.Snap+path,
+		env.SnapData+path)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func installDevices() error {
+	devicesDir := "/config/device-usb-camera/res/devices"
+
+	err := os.MkdirAll(env.SnapData+devicesDir, 0755)
+	if err != nil {
+		return err
+	}
+
+	err = hooks.CopyFile(
+		hooks.Snap+devicesDir+"/general.usb.camera.toml.example",
```

Comment:
Why are we copying an example device file when it won't actually be imported unless someone manually renames the file?

Reply:
This is to maintain consistency among different device services, where configuration files are available in the snap data directories (examples or not). It will be hard for a user to find the example general device if it is not copied over to snap data.